easybuild/easyblocks/c/clang_aomp.py

```diff
@@ -0,0 +1,276 @@
+        device_lib_dir_pattern = os.path.join(self.builddir, 'ROCm-Device-Libs-*')
+        hits = glob.glob(device_lib_dir_pattern)
+        if len(hits) == 1:
+            self.device_lib_path = hits[0]
+        else:
+            raise EasyBuildError("Could not find 'ROCm-Device-Libs' source directory in %s", self.builddir)
+
+    def configure_step(self):
+        """
+        Go through each component and setup configuration for the later Bundle install step
+        """
+        super(EB_Clang_minus_AOMP, self).configure_step()
+
+        self.prepare_sources()
+
+        num_comps = len(self.cfg['components'])
+        for idx, comp in enumerate(self.comp_cfgs):
+            name = comp['name']
+            msg = "configuring bundle component %s %s (%d/%d)..." % (name, comp['version'], idx + 1, num_comps)
+            print_msg(msg)
+            if name in self.cfg_method:
+                self.cfg_method[name](comp)
+                self.log.info(msg)
+            else:
+                self.log.warn("Component %s has no configure method!" % name)
+
+    def sanity_check_step(self):
+        """
+        Custom sanity check for ROCm
+        """
+        shlib_ext = get_shared_lib_ext()
+        custom_paths = {
+            'files': ['bin/clang', 'bin/lld', 'lib/libomp.%s' % shlib_ext,
+                      'lib/libomptarget.rtl.amdgpu.%s' % shlib_ext, 'lib/libomptarget.%s' % shlib_ext],
+            'dirs': ['amdgcn/bitcode', 'include/clang', 'include/lld', 'include/llvm',
+                     'lib/libdevice'],
+        }
+        custom_commands = ['clang --help', 'clang++ --help']
+
+        libs = ['aompextras', 'omptarget']
+        libdevice = os.path.join('lib', 'libdevice')
+
+        # Check that all AMD GFX libraries were built
+        for gfx in self.amd_gfx_archs:
+            custom_paths['files'].extend([os.path.join(libdevice, 'lib%s-amdgcn-%s.bc' % (x, gfx)) for x in libs])
+
+        # Check that CPU target OpenMP offloading library was built
+        arch = get_cpu_architecture()
+
+        # Check architecture explicitly since Clang uses potentially different names
+        if arch == X86_64:
+            arch = 'x86_64'
+        elif arch == POWER:
+            arch = 'ppc64'
+        elif arch == AARCH64:
+            arch = 'aarch64'
+        else:
+            print_warning("Unknown CPU architecture (%s) for OpenMP offloading!" % arch)
+
+        custom_paths['files'].append(os.path.join('lib', 'libomptarget.rtl.%s.%s' % (arch, shlib_ext)))
+
+        # If CUDA offloading support was requested, check that correct omptarget was built
+        if get_software_root('CUDA'):
+            custom_paths['files'].append(os.path.join('lib', 'libomptarget.rtl.cuda.%s' % shlib_ext))
+
+            for arch in self.cuda_archs:
+                sm_arch = 'sm_%s' % arch
+                custom_paths['files'].extend([os.path.join(libdevice, 'lib%s-nvptx-%s' % (x, sm_arch)) for x in libs])
```

Comment:
This looks incorrect, there's no `lib*nvptx* in `lib/libdevice` when building with CUDA support, but there's a `lib/libomptarget-nvptx*.bc` for every target CUDA compute capability.

fixed in https://github.com/nordmoen/easybuild-easyblocks/pull/5